src/User/MysqlToggleStatusModifierService.php

```diff
@@ -80,174 +95,135 @@ private function toggleStatusUnset( $toggleStatus )
+    private function insertPolicyRecord( StringifyableTable $policyTable, $toggleIdentifier, $identity, $isActive )
     {
-        $queryBuilder
-            ->setParameter( 0, $isActive )
-            ->setParameter( 1, $toggleIdentifier )
-            ->setParameter( 2, $groupIdentifier );
-        $queryBuilder->execute();
+        $identityColumn = $this->getIdentityColumnByPolicyTable( $policyTable );
+        return $this->connection->insert(
+                (string)$policyTable,
+                [
+                        "toggle_id" => $toggleIdentifier,
+                        $identityColumn => $identity,
+                        "active" => $isActive ? 1 : 0
+                ]
+        );
     }
 
     /**
-     * @param string $toggleIdentifier
-     * @param string $userIdentifier
+     * @param StringifyableTable $policyTable
+     * @param int $toggleIdentifier
+     * @param string $identity
+     * @param bool $isActive
+     * @return int
      */
-    private function tryInsertElseUpdateUserToggleToActiveState( $toggleIdentifier, $userIdentifier )
+    private function updatePolicyRecord( StringifyableTable $policyTable, $toggleIdentifier, $identity, $isActive )
     {
-        try {
-            $this->insertActiveUserActivatedToggle( $toggleIdentifier, $userIdentifier );
-        } catch ( \Exception $e ) {
-            $queryBuilder = $this->generateQueryBuilderForToggleUpdate( '`user_policy`', 'user_id' );
-            $this->updateToggle( $queryBuilder, $toggleIdentifier, $userIdentifier, true );
-        }
+        $identityColumn = $this->getIdentityColumnByPolicyTable( $policyTable );
+        return $this->connection->update(
+                (string)$policyTable,
+                [ "active" => $isActive ? 1 : 0 ],
+                [
+                        "toggle_id" => $toggleIdentifier,
+                        $identityColumn => $identity
+                ]
+        );
     }
 
     /**
-     * @param string $toggleIdentifier
-     * @param string $groupIdentifier
-     * @param bool $isActive
+     * @param StringifyableTable $policyTable
+     * @param int $toggleIdentifier
+     * @param string $identity
+     * @param boolean $isActive
+     * @return int
      */
-    private function tryInsertElseUpdateGroupToggleToAGivenState( $toggleIdentifier, $groupIdentifier, $isActive )
+    private function insertOrUpdateTogglePolicy( StringifyableTable $policyTable, $toggleIdentifier, $identity, $isActive )
     {
-        try {
-            $this->insertGroupActivatedToggle( $toggleIdentifier, $groupIdentifier, $isActive );
-        } catch ( \Exception $e ) {
-            $queryBuilder = $this->generateQueryBuilderForToggleUpdate( '`group_policy`', 'group_id' );
-            $this->updateToggle( $queryBuilder, $toggleIdentifier, $groupIdentifier, $isActive );
+        $policyRecordExists = $this->policyRecordExists( $policyTable, $toggleIdentifier, $identity );
```

Comment:
Still, maybe it was a correct implementation as there can be that a row is inserted after this line on a separate request